vmm/src/config.rs

```diff
@@ -1068,6 +1077,24 @@ impl RateLimiterGroupConfig {
     }
 }
 
+fn parse_addr(
+    s: &str,
+) -> std::result::Result<(u8 /* device */, u8 /* function */), ValidationError> {
+    let (device_str, function_str) = s
+        .split_once('.')
+        .ok_or(ValidationError::PciBdfSyntax(s.to_owned()))?;
+
+    let device_id = u8::from_str_radix(device_str.strip_prefix("0x").unwrap_or(device_str), 16)
+        .map_err(|_| ValidationError::PciBdfHex(s.to_owned()))?;
```

Comment:
I'd prefer to keep the inner error.


```suggestion
        .map_err(|e| ValidationError::PciBdfHex(e, s.to_owned()))?;
```

than you could also have a `#[source] ParseErrorType` in the enum variant definition